custom/icds/location_rationalization/views.py

```diff
@@ -1,120 +1,115 @@
 from django.contrib import messages
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
 from django.utils.decorators import method_decorator
 from django.utils.functional import cached_property
 from django.utils.translation import ugettext as _
-from django.utils.translation import ugettext_noop
+from django.utils.translation import ugettext_lazy, ugettext_noop
+from django.views.decorators.http import require_GET
 
 from memoized import memoized
 
 from corehq import toggles
 from corehq.apps.domain.views.base import BaseDomainView
+from corehq.apps.hqwebapp.utils import get_bulk_upload_form
 from corehq.apps.locations.models import LocationType
+from corehq.apps.locations.permissions import require_can_edit_locations
+from corehq.apps.locations.views import LocationsListView
 from corehq.util.files import safe_filename_header
-from corehq.util.workbook_json.excel import get_workbook
+from corehq.util.workbook_json.excel import get_workbook, WorkbookJSONError
 from custom.icds.location_rationalization.download import (
     RequestTemplateDownload,
 )
 from custom.icds.location_rationalization.dumper import Dumper
-from custom.icds.location_rationalization.forms import (
-    LocationRationalizationValidateForm,
-    LocationRationalizationTemplateForm,
-)
 from custom.icds.location_rationalization.parser import Parser
 
 
 @method_decorator([toggles.LOCATION_RATIONALIZATION.required_decorator()], name='dispatch')
-class BaseLocationRationalizationView(BaseDomainView):
-    section_name = ugettext_noop("Location Rationalization")
-
-    @property
-    def page_context(self):
-        context = {}
-        return context
-
+@method_decorator(require_can_edit_locations, name='dispatch')
+class LocationRationalizationView(BaseDomainView):
+    section_name = ugettext_lazy("Locations")
 
-class ValidateView(BaseLocationRationalizationView):
-    page_title = _('Validate')
-    urlname = 'validate_location_rationalization'
-    template_name = 'location_rationalization/validate.html'
+    page_title = _('Location Rationalization')
+    urlname = 'location_rationalization'
+    template_name = 'icds/location_rationalization.html'
 
     def section_url(self):
-        return self.page_url
+        return reverse(LocationsListView.urlname, args=[self.domain])
 
     @property
     def page_context(self):
         context = super().page_context
-        context['form'] = self.form
+        context.update({
+            'bulk_upload': {
+                "download_url": reverse('download_location_rationalization', args=[self.domain]),
+                "adjective": _("locations"),
+                "plural_noun": _("location operations"),
+                "help_link": "TODO",
```

Comment:
@mkangia Have you started docs for this feature?

Reply:
Not yet.